src/lib/genesis_ledger_helper/genesis_ledger_helper.ml

```diff
@@ -127,6 +144,21 @@ module Ledger = struct
       let local_path = Cache_dir.s3_install_path ^/ filename in
       match%bind Cache_dir.load_from_s3 [ s3_path ] [ local_path ] ~logger with
       | Ok () ->
+          let s3_hash =
```

Comment:
Maybe we could run this check before downloading the ledger?

Reply:
Yes indeed